src/lib/smart-manager.js

```diff
@@ -148,17 +148,8 @@ export function updateRootContext (context) {
  * @param {Element} component
  */
 function connectComponent (container, definition, component) {
-
-  const context$ = new Subject();
-  const disconnectionController = new AbortController();
-
-  if (component[META] == null) {
-    component[META] = new Map();
-  }
-  component[META].set(definition, { context$, disconnectionController });
   component[LAST_CONTEXT] = {};
-
-  definition.onConnect(container, component, context$, disconnectionController.signal);
+  definition.onConnect?.(container, component);
```

Comment:
I've never used the optional chaining for this kind of case so I might have missed the point but it feels like  our 3 lifecycle hooks (`onConnect`, etc., not sure how to call these) are all mandatory for our low level smart manager to work properly.

If so, I would avoid using the optional chaining and let it fail if we forget to provide one of the hooks in our smart manager observables / magic implementations.
This way it fails with an explicit error instead of silently not working ?